leakpro/attacks/mia_attacks/HSJ.py

```diff
@@ -274,78 +296,218 @@ def sanity_check(self: Self, shadow_distances_in: np.ndarray, shadow_distances_o
-                                                    image_target = self.image_target,
-                                                    initial_num_evals = self.initial_num_evals,
-                                                    max_num_evals = self.max_num_evals,
-                                                    stepsize_search = self.stepsize_search,
-                                                    num_iterations = self.num_iterations,
-                                                    gamma = self.gamma,
-                                                    constraint = self.constraint,
-                                                    batch_size = self.batch_size,
-                                                    verbose = self.verbose,
-                                                    clip_min = self.clip_min,
-                                                    clip_max = self.clip_max,
-                                                    logger = self.logger
-                                                    )
-        _ , out_data_distances = self.signal(model = self.target_model,
-                                            data_loader = target_test_loader,
-                                            norm = self.norm,
-                                            y_target = self.y_target,
-                                            image_target = self.image_target,
-                                            initial_num_evals = self.initial_num_evals,
-                                            max_num_evals = self.max_num_evals,
-                                            stepsize_search = self.stepsize_search,
-                                            num_iterations = self.num_iterations,
-                                            gamma = self.gamma,
-                                            constraint = self.constraint,
-                                            batch_size = self.batch_size,
-                                            verbose = self.verbose,
-                                            clip_min = self.clip_min,
-                                            clip_max = self.clip_max,
-                                            logger = self.logger
+        aux_data_size = int(len(target_train_indices)*self.reproducing_paper_results)
+        attack_in_indices = np.random.choice(target_train_indices, aux_data_size, replace=False)
+        attack_out_indices = np.random.choice(target_test_indices, aux_data_size, replace=False)
+
+        attack_in_dataset = self.population.subset(attack_in_indices)
+        attack_out_dataset = self.population.subset(attack_out_indices)
+
+        attack_in_dataloader = DataLoader(attack_in_dataset, batch_size=self.batch_size, shuffle=True)
+        attack_out_dataloader = DataLoader(attack_out_dataset, batch_size=self.batch_size, shuffle=True)
+
+        _, in_data_distances = self.signal( self.target_model,
```

Comment:
Here, the target is used as input on the actual target data. Why did we use shadow models in the attack?